### ISSUE TITLE
💣  Unable to start VM: WARNING: The vboxdrv kernel module is not loaded

### ISSUE THREAD

**Issue by pooja-kulal** [kubernetes/minikube#4060]
ubuntu@knode:~$ sudo minikube start
[sudo] password for ubuntu: 
😄  minikube v1.0.0 on linux (amd64)
🤹  Downloading Kubernetes v1.14.0 images in the background ...
🔥  Creating virtualbox VM (CPUs=2, Memory=2048MB, Disk=20000MB) ...

💣  Unable to start VM: create: precreate: We support Virtualbox starting with version 5. Your VirtualBox install is "WARNING: The vboxdrv kernel module is not loaded. Either there is no module\n         available for the current kernel (4.8.0-36-generic) or it failed to\n         load. Please recompile the kernel module and install it by\n\n           sudo /sbin/vboxconfig\n\n         You will not be able to start VMs until this problem is fixed.\n6.0.4r128413". Please upgrade at https://www.virtualbox.org

😿  Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
👉  https://github.com/kubernetes/minikube/issues/new

**Comment by pooja-kulal** [comment]
VM version 6.0.4
ubuntu 16.04

**Comment by whoan** [comment]
Did you try loading the module manually? `sudo modprobe vboxdrv`

**Comment by pooja-kulal** [comment]
yes i tried . but i m getting an error.

modprobe: ERROR: could not insert 'vboxdrv': Required key not available

**Comment by tstromberg** [comment]
Closing as dupe of #4043 

Basically, this error is a sign of a failed VirtualBox installation. The solution is to reinstall VirtualBox. We should direct users to do so.

**Comment by whoan** [comment]
In my case, it was because of not loaded modules after a new VirtualBox installation. You might direct users to reboot first if VirtualBox was installed recentrly, and if the problem persists, reinstall VBox.

**Comment by adarshaJha** [comment]
still gettting the same error : 

```
 minikube v1.5.1 on Ubuntu 16.04
🔥  Creating virtualbox VM (CPUs=2, Memory=2000MB, Disk=20000MB) ...
🔄  Retriable failure: create: precreate: We support Virtualbox starting with version 5. Your VirtualBox install is "WARNING: The vboxdrv kernel module is not loaded. Either there is no module\n         available for the current kernel (4.15.0-66-generic) or it failed to\n         load. Please recompile the kernel module and install it by\n\n           sudo /sbin/vboxconfig\n\n         You will not be able to start VMs until this problem is fixed.\n6.0.14r133895". Please upgrade at https://www.virtualbox.org
🔥  Creating virtualbox VM (CPUs=2, Memory=2000MB, Disk=20000MB) ...
🔄  Retriable failure: create: precreate: We support Virtualbox starting with version 5. Your VirtualBox install is "WARNING: The vboxdrv kernel module is not loaded. Either there is no module\n         available for the current kernel (4.15.0-66-generic) or it failed to\n         load. Please recompile the kernel module and install it by\n\n           sudo /sbin/vboxconfig\n\n         You will not be able to start VMs until this problem is fixed.\n6.0.14r133895". Please upgrade at https://www.virtualbox.org

```
has anyone really solved it ?

**Comment by tstromberg** [comment]
@adarshaJha - Yes. I suggest you try running `sudo modprobe vboxdrv`, and if that fails, `sudo /sbin/vboxconfig` and try `sudo modprobe vboxdrv` again.

Worst case, I would try reinstalling VirtualBox and rebooting.

**Comment by Chitrank-Dixit** [comment]
When I am trying to start minikube 

`$ minikube start`

I get the below error

```
E1108 17:29:11.547446     435 start.go:174] Error starting host: Error creating host: Error executing step: Running precreate checks.
: We support Virtualbox starting with version 5. Your VirtualBox install is "WARNING: The vboxdrv kernel module is not loaded. Either there is no module\n         available for the current kernel (5.0.0-31-generic) or it failed to\n         load. Please recompile the kernel module and install it by\n\n           sudo /sbin/vboxconfig\n\n         You will not be able to start VMs until this problem is fixed.\n6.0.12r133076". Please upgrade at https://www.virtualbox.org.
```

when I try to fix this by trying the command

`$ sudo /sbin/vboxconfig`

I get this error

```
vboxdrv.sh: Stopping VirtualBox services.
vboxdrv.sh: Starting VirtualBox services.
vboxdrv.sh: You must sign these kernel modules before using VirtualBox:
  vboxdrv vboxnetflt vboxnetadp vboxpci
See the documenatation for your Linux distribution..
vboxdrv.sh: Building VirtualBox kernel modules.
vboxdrv.sh: Failed to enroll secure boot key..
vboxdrv.sh: failed: modprobe vboxdrv failed. Please use 'dmesg' to find out why.

There were problems setting up VirtualBox.  To re-start the set-up process, run
  /sbin/vboxconfig
as root.  If your system is using EFI Secure Boot you may need to sign the
kernel modules (vboxdrv, vboxnetflt, vboxnetadp, vboxpci) before you can load
them. Please see your Linux system's documentation for more information.
```

I have already reinstalled virtualbox, but I think that is not helping.

**Comment by adarshaJha** [comment]
@tstromberg  i am still facing the issue, after doing what you suggested. Is there anything else i should try.

**Comment by AlekseyKanaev** [comment]
@adarshaJha maybe you should try to use kvm instead of virtualbox? I faced the same problem with virtualbox, but installing kvm and using it instead did the trick.

**Comment by prashant-shahi** [comment]
Reinstalling Virtualbox worked out for me.

**Comment by ishika261** [comment]
> When I am trying to start minikube
> 
> `$ minikube start`
> 
> I get the below error
> 
> ```
> E1108 17:29:11.547446     435 start.go:174] Error starting host: Error creating host: Error executing step: Running precreate checks.
> : We support Virtualbox starting with version 5. Your VirtualBox install is "WARNING: The vboxdrv kernel module is not loaded. Either there is no module\n         available for the current kernel (5.0.0-31-generic) or it failed to\n         load. Please recompile the kernel module and install it by\n\n           sudo /sbin/vboxconfig\n\n         You will not be able to start VMs until this problem is fixed.\n6.0.12r133076". Please upgrade at https://www.virtualbox.org.
> ```
> 
> when I try to fix this by trying the command
> 
> `$ sudo /sbin/vboxconfig`
> 
> I get this error
> 
> ```
> vboxdrv.sh: Stopping VirtualBox services.
> vboxdrv.sh: Starting VirtualBox services.
> vboxdrv.sh: You must sign these kernel modules before using VirtualBox:
>   vboxdrv vboxnetflt vboxnetadp vboxpci
> See the documenatation for your Linux distribution..
> vboxdrv.sh: Building VirtualBox kernel modules.
> vboxdrv.sh: Failed to enroll secure boot key..
> vboxdrv.sh: failed: modprobe vboxdrv failed. Please use 'dmesg' to find out why.
> 
> There were problems setting up VirtualBox.  To re-start the set-up process, run
>   /sbin/vboxconfig
> as root.  If your system is using EFI Secure Boot you may need to sign the
> kernel modules (vboxdrv, vboxnetflt, vboxnetadp, vboxpci) before you can load
> them. Please see your Linux system's documentation for more information.
> ```
> 
> I have already reinstalled virtualbox, but I think that is not helping.

have you found out any solution for this problem?? i am still stuck !

**Comment by rickdemarte** [comment]
I had the same issue, minikube said me to run **sudo /sbin/vboxconfig** and I did.
I got a error in "Building VirtualBox kernel modules" so I had a look on **/var/log/vbox-setup.log**
In my case, the package **gcc-12** was not installed.
In your case, the error is diferent, so maybe take a look in the same log, you may see if there is something missing.

Installing gcc-12 worked for me.
